File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/DeviceVerification.java
package de.tum.in.tumcampusapp.models.tumcabe;

import android.content.Context;

import java.math.BigInteger;
import java.security.SecureRandom;
import java.util.Date;

import de.tum.in.tumcampusapp.auxiliary.AuthenticationManager;
import de.tum.in.tumcampusapp.exceptions.NoPrivateKey;

public class DeviceVerification {

    private String signature;
    private String date;
    private String rand;
    private String device;

    public DeviceVerification(Context c) throws NoPrivateKey {
        //Create some data
        this.date = (new Date()).toString();
        this.rand = new BigInteger(130, new SecureRandom()).toString(32);
        this.device = AuthenticationManager.getDeviceID(c);

        //Sign this data for verification
        AuthenticationManager am = new AuthenticationManager(c);
        this.signature = am.sign(date + rand + this.device);
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/DeviceVerification.kt
package de.tum.`in`.tumcampusapp.models.tumcabe

import android.content.Context
import de.tum.`in`.tumcampusapp.auxiliary.AuthenticationManager
import de.tum.`in`.tumcampusapp.exceptions.NoPrivateKey
import java.math.BigInteger
import java.security.SecureRandom
import java.util.*

data class DeviceVerification(var signature: String = "",
                              var date: String = "",
                              var rand: String = "",
                              var device: String = "") {
    companion object {
        @Throws(NoPrivateKey::class)
        fun getDeviceVerification(c: Context): DeviceVerification {
            //Create some data
            val date = Date().toString()
            val rand = BigInteger(130, SecureRandom()).toString(32)
            val device = AuthenticationManager.getDeviceID(c)

            return DeviceVerification(date = date, rand = rand, device = device,
                    //Sign this data for verification
                    signature = AuthenticationManager(c).sign(date + rand + device))
        }
    }
}
